### PR TITLE
Allow manual editing of browser extension connection keys & add option for custom extension ID

### LINKED ISSUE #3819: Allow manual entry/editing of KeePassXC-Browser keys
This is a FR based on an IRC chat with @varjolintu 

Essentially before they Key/Value bits were migrated from being an entry in the 'Root' folder to being a custom data section of the KDBX file (and thus available in 'Browser Integration' under Database settings...').  You could at least manually modify the existing Key/Values and also add in ones manually.

This is handy if you've got multiple PCs and might want to reuse a key value after a re-install or just want to better manage this data as you're testing newer versions of the extension, etc...

### PR DESCRIPTION
**Pull request by varjolintu**
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (non-breaking change which adds functionality)

## Description and Context
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )
Allows editing the browser extension connection keys. Both key and identifier value can be changed. Editing "Created" timestamp is not allowed.

Editing works using double click on the mouse. A keyboard shortcut could be also added if needed.

Also adds an option to use a custom extension ID with debug builds.

Fixes #3819.

## Screenshots
Screen Shot 2020-04-13 at 11 16 19

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Manually.

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

**Comment by droidmonkey** [comment]
Along with this, in debug builds **only**, can we have the ability to add a custom extension ID that will be output in the native messaging json file?

**Comment by varjolintu** [comment]
@droidmonkey That would be a really nice addition. I'm already tired of editing the JSON files when debugging.

**Comment by varjolintu** [comment]
@droidmonkey The feature has been added to the PR.

File: src/browser/BrowserOptionDialog.cpp [hunk 1]
```diff
@@ -159,6 +167,8 @@ void BrowserOptionDialog::loadSettings()
     m_ui->browserGlobalWarningWidget->setCloseButtonVisible(false);
     m_ui->browserGlobalWarningWidget->setAutoHideTimeout(-1);
 #endif
+    m_ui->useCustomExtensionId->setChecked(settings->useCustomExtensionId());
+    m_ui->customExtensionId->setText(settings->customExtensionId());
```
Comment (droidmonkey) [inline src/browser/BrowserOptionDialog.cpp:171]: Should wrap these in `#ifdef QT_DEBUG`

File: src/browser/BrowserOptionDialog.cpp [hunk 2]
```diff
@@ -198,6 +208,8 @@ void BrowserOptionDialog::saveSettings()
     settings->setSearchInAllDatabases(m_ui->searchInAllDatabases->isChecked());
     settings->setSupportKphFields(m_ui->supportKphFields->isChecked());
     settings->setNoMigrationPrompt(m_ui->noMigrationPrompt->isChecked());
+    settings->setUseCustomExtensionId(m_ui->useCustomExtensionId->isChecked());
+    settings->setCustomExtensionId(m_ui->customExtensionId->text());
```
Comment (droidmonkey) [inline src/browser/BrowserOptionDialog.cpp:212]: Same here

File: src/browser/BrowserOptionDialog.cpp (hunk 1 above)
Comment (varjolintu) [reply src/browser/BrowserOptionDialog.cpp:173]: Fixed.

**Pull request merged**